src/util/kaldi-thread.h

```diff
@@ -181,11 +181,21 @@ class TaskSequencer {
     KALDI_ASSERT((config.num_threads_total <= 0 ||
                   config.num_threads_total >= config.num_threads) &&
                  "num-threads-total, if specified, must be >= num-threads");
+    if (config.num_threads == 0) {
```

Comment:
I think it would be nicer if instead of synchronous_, you added a member num_threads_, which was just a copy of config.num_threads.  There's a chance we might need that later for something else anyway.